File: array_easy_2.cpp
//for 用的太多，现在都开始不喜欢写i++了。。。
//https://leetcode.com/explore/featured/card/top-interview-questions-easy/92/array/564/

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        int sum = 0;
        int i = 1;
        while(i < n){
            if(prices[i - 1] < prices[i]) sum += prices[i] - prices[i - 1];
            i++;
        }
        return sum;
    }
};

//之前python版本的判断小范围数据可以不要

File: array_easy_2.py
#https://leetcode.com/explore/featured/card/top-interview-questions-easy/92/array/564/
#python 手打indexed block 需要非常注意，没有分号，有冒号，我太难了

class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        length = len(prices)
        if length < 2: return 0
        max_profit = 0
        for i in range(1, length):
            if prices[i - 1] < prices[i]:
                max_profit += prices[i] -  prices[i - 1]
        return max_profit
  
  #小规模问题，如果单独考虑length = 0, length = 1， 可以少很多麻烦
